core/src/main/java/io/grpc/ClientInterceptors.java

```diff
@@ -115,70 +115,102 @@ private ProcessInterceptorChannel(Channel channel, Iterable<ClientInterceptor> i
   /**
    * A {@link Call} which forwards all of it's methods to another {@link Call}.
    */
-  public static class ForwardingCall<ReqT, RespT> extends Call<ReqT, RespT> {
+  public abstract static class ForwardingCall<ReqT, RespT> extends Call<ReqT, RespT> {
 
-    private final Call<ReqT, RespT> delegate;
-
-    public ForwardingCall(Call<ReqT, RespT> delegate) {
-      this.delegate = delegate;
-    }
+    /**
+     * Returns the delegated {@code Call}
+     */
+    protected abstract Call<ReqT, RespT> delegate();
 
     @Override
     public void start(Listener<RespT> responseListener, Metadata.Headers headers) {
-      this.delegate.start(responseListener, headers);
+      this.delegate().start(responseListener, headers);
     }
 
     @Override
     public void request(int numMessages) {
-      this.delegate.request(numMessages);
+      this.delegate().request(numMessages);
     }
 
     @Override
     public void cancel() {
-      this.delegate.cancel();
+      this.delegate().cancel();
     }
 
     @Override
     public void halfClose() {
-      this.delegate.halfClose();
+      this.delegate().halfClose();
     }
 
     @Override
     public void sendPayload(ReqT payload) {
-      this.delegate.sendPayload(payload);
+      this.delegate().sendPayload(payload);
     }
   }
 
+  private static final Call<Object, Object> NOOP_CALL = new Call<Object, Object>() {
```

Comment:
I didn't intend to make this no-op ``Call`` a general-purpose base class for overriding, because I thought we were not feeling completely comfortable about having such a base class. Otherwise I would just change the base ``Call`` class.

Now as I think about this solution again, I don't find it appealing enough. Every interceptor implementation that may fail in ``start()`` needs to add a ``startFailed`` flag and have a nearly identical ``delegate()``. It is repetitive and leaves quite a room for anti-pattern, i.e., just not doing exactly so.

While we can keep ``abstract delegate()`` for the flexibility, maybe we can provide a convenient class like:
```java
CheckedForwardingCall extends ForwardingCall {
  final start(...);
  final delegate(...);
  abstracted checkedStart() throws Exception;
}
```